notificationworkerlambda/src/main/scala/com/gu/notifications/worker/tokens/TokenService.scala

```diff
@@ -9,12 +9,13 @@ import db.{HarvestedToken, RegistrationService, Topic}
 import fs2.Stream
 import play.api.libs.json.{Format, Json}
 
+import java.time.Instant
 import scala.concurrent.ExecutionContextExecutor
 
 case class IndividualNotification(notification: Notification, token: String)
 case class BatchNotification(notification: Notification, token: List[String])
 
-case class ChunkedTokens(notification: Notification, tokens: List[String], range: ShardRange) {
+case class ChunkedTokens(notification: Notification, tokens: List[String], range: ShardRange, notificationAppReceivedTime: Option[Instant]) {
```

Comment:
This is optional for now so that any messages which are already in the queue (at deployment time) can still be processed; we'll be able to make this mandatory in a follow-up PR.